File: Python/1299.py
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:

        # Solution 1
        # Max Method

        # base case
        if len(arr) == 0:
            return arr

        rightMax = -1

        for i in range(len(arr)-1, -1, -1):
            temp = max(rightMax, arr[i])
            arr[i] = rightMax
            rightMax = temp

        return arr

File: Java/1299.java
class Solution{
    public int[] replaceElements(int[] arr) {
        
        // Solution 1

        // base case
        if (arr.length){
            return arr;
        }

        int rightmax = -1;

        for (int i = arr.length; i >= 0; i--){
            temp = Math.max(rightmax, arr[i]);
            arr[i] = rightmax;
            rightmax = temp
        }
        return arr;
    }
}
